### PR TITLE
iso8601 spec

### PR DESCRIPTION
**Pull request by ahorek**

### CODE REVIEW

File: library/date/iso8601_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+require_relative '../../spec_helper'
+require 'date'
+
+describe "Date#iso8601" do
+  class StringSubclass < String; end
```
Comment (eregon) [inline library/date/iso8601_spec.rb]: Could you use `Class.new(String)` instead, where it is used below?
Declaring it like this means it's defining `StringSubclass` at the top-level.

File: library/date/iso8601_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+require_relative '../../spec_helper'
+require 'date'
+
+describe "Date#iso8601" do
```
Comment (eregon) [inline library/date/iso8601_spec.rb]: `Date.iso8601` since these are specs for the class method.

**Comment by ahorek** [comment]
@eregon thanks, fixed

File: library/date/iso8601_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    d.should == Date.civil(-4712, 1, 1)
+  end
+
+  it "parses a Symbol into a Date object" do
+    d = Date.iso8601(:'2015-10-15')
+    d.should == Date.civil(2015, 10, 15)
+  end
+
+  it "parses a StringSubclass into a Date object" do
+    StringSubclass = Class.new(String)
```
Comment (eregon) [inline library/date/iso8601_spec.rb]: This still assigns a constant at the top-level, can you make it a local variable instead?
Alternatively, you could also use mocks like `object = mock("str")`, `object.should_receive(:to_str).and_return("-4712-01-01")`, but then it doesn't really test a subclass instance, it's more general.

Comment (eregon) [reply library/date/iso8601_spec.rb]: I should probably add a check in CI that no unexpected top-level constant is added.
Might be tricky to implement though.

**Pull request merged**